R/test-app.R

```diff
@@ -128,20 +134,92 @@ test_app <- function(
     }
   }
 
-  is_currently_testing <- testthat::is_testing()
 
-  ret <- testthat::test_dir(
+
+  if (testthat::is_testing()) {
+    # Normalize the reporter given any input
+    outer_reporter <- testthat::with_reporter(reporter, testthat::get_reporter(), start_end_reporter = FALSE)
+
+    outer_context <- outer_reporter$.context
+
+    # If a test is currently active, stop it and restart on exit
+    test_name <- NULL
+    if (inherits(outer_reporter, "MultiReporter")) {
+      # Find the SnapshotReporter, as the `test` value is available
+      snapshot_reporters <- Filter(outer_reporter$reporters, f = function(x) inherits(x, "SnapshotReporter"))
+      if (length(snapshot_reporters) > 0) {
+        # TODO-barret; Should an error be thrown here instead?
+        # "Please execute `test_app()` outside of your `test_that()` call" ?
```

Comment:
Should we throw an error if currently testing? The progress reporter visually has trouble with exiting a test and restarting a test.

![Screen Shot 2022-04-28 at 4 51 38 PM](https://user-images.githubusercontent.com/93231/165843573-c541a976-eea9-4f45-aa12-6849596e69b7.png)

Reply:
It works as is. Leaving for now. 

When called outside of a `test_that()` call. Output is displayed as expected.